src/ddscxx/src/org/eclipse/cyclonedds/core/MiscUtils.cpp

```diff
@@ -18,6 +18,10 @@
 
 #include "dds/dds.h"
 
+#ifdef _WIN32
+#pragma warning(disable : 4996)
+#endif
+
```

Comment:
I'm wondering what is deprecated that we're still using? If it's something that's marked deprecated in Cyclone DDS, we probably want to use the alternative instead.

Reply:
The main issue is that the errors that are being suppressed is that we are using functions like fopen, which for windows is superceded by functions like fopen_s, but since these functions are not present in linux, i think we should still use fopen and the like and suppress these errors in windows compiles.

Reply:
In this case it maybe because of `strncpy`? We have a `ddsrt_strlcpy` available for that. I don't mind use of `fopen`, but for functions where we have better alternatives in e.g. `ddsrt`, I think it's better to replace the instance. Also, for these `#pragma`'s it may be better to use the `DDSRT_MSVC_ON` macro's.

Reply:
I will look into which of the "unsafe" functions can be replaced by their `ddsrt` versions.
I will also check whether the `DDSRT_MSVC_ON` macro's can replace the `pragma` statements